ext/drreg/drreg.c

```diff
@@ -105,13 +154,30 @@ typedef struct _reg_info_t {
 
 #define GPR_IDX(reg) ((reg)-DR_REG_START_GPR)
 
+/* The applicable register range is what's used internally to iterate over all possible
+ * SIMD registers for a given build. Regs are resized to zmm when testing via SIMD_IDX().
+ */
+#ifdef SIMD_SUPPORTED
+#    define DR_REG_APPLICABLE_START_SIMD DR_REG_START_ZMM
+#    define DR_REG_APPLICABLE_STOP_SIMD DR_REG_STOP_ZMM
+#    define SIMD_IDX(reg) ((reg_resize_to_opsz(reg, OPSZ_64)) - DR_REG_START_ZMM)
```

Comment:
These also seem x86-specific, not SIMD_SUPPORTED-specific, right?  Should we add DR_REG_START_SIMD (can get complex I guess: the widest if there are subregs like on x86 and IIRC ARM)?